runtime/expr/function/ip.go

```diff
@@ -30,15 +30,22 @@ func (n *NetworkOf) Call(ctx zed.Allocator, args []zed.Value) *zed.Value {
 		}
 	} else {
 		// two args
-		id := args[1].Type.ID()
 		body := args[1].Bytes
-		if id == zed.IDNet {
+		switch id := args[1].Type.ID(); {
+		case id == zed.IDNet:
 			cidrMask := zed.DecodeNet(body)
 			if !bytes.Equal(cidrMask.IP, cidrMask.Mask) {
 				return newErrorf(n.zctx, ctx, "network_of: network arg not a cidr mask")
 			}
 			mask = cidrMask.Mask
-		} else if zed.IsInteger(id) {
+		case id == zed.IDIP:
+			ip := zed.DecodeIP(body)
+			if ip.Is6() {
+				return newErrorf(n.zctx, ctx, "network_of: mask cannot be an IPv6 address")
+			}
+			b := ip.As4()
+			mask = net.IPv4Mask(b[0], b[1], b[2], b[3])
```

Comment:
Can't this just be...?
```
mask = net.IPMask(zed.DecodeIP(body).AsSlice())
```
Also, I noticed that allows a non-contiguous mask into a zed network value which then formats it in a non-ZSON compatible form.  So maybe we should detect non-contiguous masks?